src/parse-result.js

```diff
@@ -63,6 +66,14 @@ export function namespace(options) {
     }
   }
 
+  [BooleanElement, StringElement, NumberElement, ObjectElement, ArrayElement].forEach((item) => {
```

Comment:
Instead of doing this for every element, can't we just add the `sourceMap` accessor to the base element?